backend/remote-state/http/backend.go

```diff
@@ -0,0 +1,334 @@
+	"time"
+
+	"github.com/hashicorp/terraform/backend"
+	"github.com/hashicorp/terraform/helper/schema"
+)
+
+// New Backend
+func New() backend.Backend {
+	s := &schema.Backend{
+		Schema: map[string]*schema.Schema{
+			"address": {
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: "(Required) The address of the REST endpoint.",
+			},
+
+			"update_method": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Default:     "POST",
+				Description: "(Optional) HTTP method to use when updating state. Defaults to POST.",
+			},
+
+			"lock_address": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Description: "(Optional) The address of the lock REST endpoint. Defaults to address.",
+			},
+
+			"lock_method": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Default:     "LOCK",
+				Description: "(Optional) The HTTP method to use when locking. Defaults to LOCK.",
+			},
+
+			"unlock_address": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Description: "(Optional) The address of the unlock REST endpoint. Defaults to address.",
+			},
+
+			"unlock_method": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Default:     "UNLOCK",
+				Description: "(Optional) The HTTP method to use when unlocking. Defaults to UNLOCK.",
+			},
+
+			"username": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Description: "(Optional) The username for HTTP basic authentication.",
+			},
+
+			"password": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Description: "(Optional) The password for HTTP basic authentication.",
+			},
+
+			"skip_cert_verification": {
+				Type:        schema.TypeBool,
+				Optional:    true,
+				Default:     false,
+				Description: "(Optional) Whether to skip TLS verification. Defaults to false.",
+			},
+
+			"local_cert_ca_file": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Description: "CA to use when the rest api is using a self signed certificate.",
+			},
+
+			"mutual_tls_authentication": {
+				Type:        schema.TypeBool,
+				Optional:    true,
+				Default:     false,
+				Description: "Use mutual tls authentication.local_cert_file, local_key_file, local_ca_file needs to be set. Defaults to false.",
+			},
+
+			"local_cert_file": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Description: "Cert file to be used for mutual tls authentication.",
+			},
+
+			"local_key_file": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Description: "Key file to be used for mutual tls authentication.",
+			},
+
+			"local_ca_file": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Description: "CA to be used for mutual tls authentication.",
+			},
+		},
+	}
+
+	result := &Backend{Backend: s}
+	result.Backend.ConfigureFunc = result.configure
+	return result
+}
+
+// Backend Structure
+type Backend struct {
```

Comment:
You should document this struct (and its attributes) more thoroughly. You might as well say: 

```
// Add x and y to calculate z
z := x + y
```